classes/Assets/Credits.php

```diff
@@ -882,51 +882,76 @@ static function checkJoinPaid($userId, $toStream, $fromStream = null, $options =
 		$fromPublisherId = Q::ifset($fromStream, "publisherId", null);
 		$fromStreamName = Q::ifset($fromStream, "streamName", null);
 
-		$reasons = array('JoinedPaidStream');
-		if (!empty($options['reasons'])) {
-			$reasons = array_merge($reasons, $options['reasons']);
-		}
-
-		// find latest credits transfer
+        $latestLeftPaidStream = 0;
+        $leftReason = 'LeftPaidStream';
+
+        // find latest date user left paid stream
+        $left_assets_credits = Assets_Credits::select()
+            ->where(array(
+                'toUserId' => $userId,
+                'toPublisherId' => $fromPublisherId,
+                'toStreamName' => $fromStreamName,
+                'fromPublisherId' => $toPublisherId,
+                'fromStreamName' => $toStreamName,
+                'reason' => $leftReason
+            ))
+            ->ignoreCache()
+            ->options(array("dontCache" => true))
+            ->orderBy('insertedTime', false)
+            ->limit(1)
+            ->fetchDbRow();
+        if ($left_assets_credits) {
+            $latestLeftPaidStream = $left_assets_credits->insertedTime;
+        }
+
+		// find all credits transfer for paid stream latest than latest left stream
 		$joined_assets_credits = Assets_Credits::select()
 		->where(array(
 			'fromUserId' => $userId,
 			'toPublisherId' => $toPublisherId,
 			'toStreamName' => $toStreamName,
 			'fromPublisherId' => $fromPublisherId,
 			'fromStreamName' => $fromStreamName,
-			'reason' => $reasons
+			'reason !=' => $leftReason,
+            'insertedTime >' => $latestLeftPaidStream
```

Comment:
I'd recommend using `>=` not `>` just in case

Reply:
All requested changes done.